recipes/nghttp3/all/conanfile.py

```diff
@@ -0,0 +1,85 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.cmake import CMake, CMakeToolchain, CMakeDeps, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import get, rmdir, copy
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class Nghttp3Conan(ConanFile):
+    name = "nghttp3"
+    description = "HTTP/2 C Library and tools"
+    topics = ("http", "http3")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://nghttp2.org/nghttp3/"
+    license = "MIT"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.20 <4]")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "6":
+            raise ConanInvalidConfiguration(f"{self.ref} requires GCC >= 6.0")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ENABLE_SHARED_LIB"] = self.options.shared
+        tc.variables["ENABLE_STATIC_LIB"] = not self.options.shared
+        tc.variables["ENABLE_LIB_ONLY"] = True
+        if is_apple_os(self):
+            # workaround for: install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
+            tc.cache_variables["CMAKE_MACOSX_BUNDLE"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
```

Comment:
```suggestion
```

The CMakeDeps is useful to generate `Findxxx.cmake` for those packages listed in `requirements()` method. Here, in `nghttp3`, it does not require dependencies, so `CMakeDeps` can be disposed, because is not used.

But what about `cmake` as dependency? CMake will be solved via PATH environment variable, configured in `conanbuildenv.sh`, but it's generated by `VirtualBuildEnv`.